src/main/java/org/commcare/formplayer/services/MediaHandler.kt

```diff
@@ -19,16 +21,37 @@ class MediaHandler(val file: MultipartFile) {
     /**
      * Saves file in the given parent directory
      */
-    fun saveFile(parentDirPath: Path): String {
+    fun saveFile(parentDirPath: Path, session: SerializableFormSession, username: String, asUser: String,
+                 domain: String, appId: String): String {
         validateFile()
         val fileId = PropertyUtils.genUUID()
         val parent = parentDirPath.toFile()
         parent.mkdirs()
         var fileIdWithExt = fileId
-        FileUtils.getExtension(file.originalFilename)?.let { fileIdWithExt = "$fileId.$it" }
-        val desintationFile = getMediaFilePath(parentDirPath, fileIdWithExt).toFile()
+        val fileExtension = FileUtils.getExtension(file.originalFilename)
+        fileExtension?.let { fileIdWithExt = "$fileId.$it" }
+        val filePath = getMediaFilePath(parentDirPath, fileIdWithExt)
+        val destinationFile = filePath.toFile()
+
+        val metaDataRecordToDelete = MediaMetaDataService().findByFilePath(filePath.toString())
+        val mediaMetaDataService = MediaMetaDataService()
+        metaDataRecordToDelete?.let{ mediaMetaDataService.deleteMetaDataById(it.id) }
+
         try {
-            FileUtils.copyFile(file.inputStream, desintationFile)
+            FileUtils.copyFile(file.inputStream, destinationFile)
+            val mediaMetaData = MediaMetadataRecord(
+                fileId,
+                filePath.toString(),
+                session,
+                fileExtension,
+                file.size.toInt(),
+                username,
+                asUser,
+                domain,
+                appId
+            )
+            mediaMetaDataService.saveMediaMetaData(mediaMetaData)
```

Comment:
we should move out this block out of `try {} catch(){}` block since that's meant to catch the IOExceptions happening explicitly while saving file on filesystem.

Reply:
I see, it makes sense to keep the `try {} catch(){}` focused on that specific exception. 
My thinking here was that we should save the metadata after the file is successfully saved, and this was the only place that would work. I can move the metadata save up above the `try`. Is there any concern about what happens with the metadata record if the file save fails, like should it be deleted in the `catch` block or something like that?

Reply:
Code below won't ever be run, right? The `try-catch` will either return fileIdWithExt or throw an exception.

Reply:
no `try` doesn't really `return` without us explicitly writing a `return` statement in the code. So in this case if there is an error the code will exit throwing an exception. Otherewise it should continue executing code written below line by line.

Reply:
There is currently a return statement in the `try` block `return fileIdWithExt` that was part of https://github.com/dimagi/formplayer/pull/1198. Should I remove that?

Reply:
Ahh yes I missed it, we should move the return out of try-catch and put it in the end of the method.